src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -12,23 +12,43 @@
 namespace Symfony\Component\Cache\Adapter;
 
 use Psr\Cache\CacheItemInterface;
-use Psr\Cache\CacheItemPoolInterface;
 use Symfony\Component\Cache\CacheItem;
 use Symfony\Component\Cache\Exception\InvalidArgumentException;
 
 /**
  * @author Nicolas Grekas <p@tchwork.com>
  */
-abstract class AbstractAdapter implements CacheItemPoolInterface
+abstract class AbstractAdapter implements AdapterInterface
 {
+    /**
+     * @var string
+     */
```

Comment:
You're not always looking at the code via your IDE. Imo those doc blocks are valuable until you can do a `private AdapterInterface $decorated;` for example.